assembly/src/parsers/ast/serde/serialization.rs

```diff
@@ -51,6 +51,19 @@ impl ByteWriter {
         Ok(())
     }
 
+    pub fn write_docs(&mut self, val: &Option<String>) {
```

Comment:
minor nit: could we put this function after `wirte_procedure_id()`. This way, the sequence of functions would be `write_procedure_name()`, `write_procedure_id()`, `write_docs()`.